File: templates/ros_node.py
#! /usr/bin/env python3
import rclpy
from rclpy.node import Node

class FirstNode(Node):
    def __init__(self):

        super().__init__("first") # initialize the node with the name "first"
        self.get_logger().info("Hello, ROS2!") # log a message to the console

def main(args=None):
    rclpy.init(args=args) # initialize the ROS2 communication
    node = FirstNode() # create an instance of the FirstNode class
    rclpy.spin(node)
    rclpy.shutdown() # shutdown the ROS2 communication


if __name__ == '__main__':
    main()

File: templates/ros_node.cpp
/*
 * This template is for ROS2 (Humble)
 * Add necessary code for full utilization
 *
 */

#include "rclcpp/rclcpp.hpp"

class Node_name: public rclcpp::Node
{
public:
    Node_name() : Node("Node_name")
    {
        RCLCPP_INFO(this->get_logger(), "Hello in CPP");
        //other initializations
    }
private:
    //callback and other defenitions
};




int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<Node_name>();
    
    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;

}
